pkg/gui/controllers/files_controller.go

```diff
@@ -627,7 +634,7 @@ func (self *FilesController) toggleTreeView() error {
 }
 
 func (self *FilesController) handleStashSave(stashFunc func(message string) error, action string, errorMsg string) error {
-	if !self.helpers.WorkingTree.IsWorkingTreeDirty() {
+	if action != self.c.Tr.Actions.StashIncludeUntrackedChanges && !self.helpers.WorkingTree.IsWorkingTreeDirty() {
```

Comment:
what's the intention behind this line?

Reply:
The method `IsWorkingTreeDirty` returns `false` if there are only untracked files, so the original expression evaluates to `true` and an error is raised. But I want to be able to stash these untracked files even if they're the only ones shown by `git status`.

Reply:
I see. Rather than have conditionally logic inside this function for the sake of one use case, I say we pull the check out to the places that actually need it. This will add some duplication but I much prefer duplication to a clunky abstraction. So in the end `handleStashSave` will only call `self.c.Prompt` and it'll be up to the calling function to handle the validation